### PR TITLE
DM-6320: Port to Python 3

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: tests/raDecToStr.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
+from builtins import range
```
Comment (PaulPrice) [inline tests/raDecToStr.py]: Commit message doesn't mention `range`.

File: python/lsst/utils/multithreading/SharedData.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def __getattribute__(self, name):
         return self._d[name]
 
     def __setattr__(self, name, value):
-        if name == "_d" or len(self._d) == 0 or name in self.__dict__.keys():
+        if name == "_d" or len(self._d) == 0 or name in self.__dict__:
```
Comment (PaulPrice) [inline python/lsst/utils/multithreading/SharedData.py]: I think "modern python" would have `self._d` instead of `len(self._d) == 0`.

File: python/lsst/utils/multithreading/SharedData.py [hunk 3]
```diff
@@ -124,19 +124,19 @@ def initData(self, data):
         """
         with self._cond:
             bad = []
-            realattrs = self.__dict__.keys()
-            for key in data.keys():
+            realattrs = list(self.__dict__.keys())
```
Comment (PaulPrice) [inline python/lsst/utils/multithreading/SharedData.py]: `set` would be better because we want to see what's `in` it.

In fact, I would write the whole `for` loop that's about to come as a `set` operation:

    bad = set(data.keys()).intersection(set(self.__dict__.keys()))

File: python/lsst/utils/tests.py [hunk 4]
```diff
@@ -22,6 +22,9 @@
 
 """Support code for running unit tests"""
 from __future__ import print_function
+from __future__ import division
```
Comment (PaulPrice) [inline python/lsst/utils/tests.py]: Is there a reason we don't put this in every file?

Any reason not to combine it with the above line (I actually like the style of one `import` per line, but that's not what we generally do).

Comment (timj) [inline python/lsst/utils/tests.py]: I believe coding policy is that these `__future__` imports should always be used (and absolute_import as well for init files) but retrospectively adding the division one usually needs some thinking.

File: python/lsst/p_lsstSwig.i [hunk 5]
```diff
@@ -53,6 +53,12 @@
 %include "carrays.i"
 %include "typemaps.i"
 
+// Allow Python2 to accept unicode strings
+%module unicode_strings
```
Comment (PaulPrice) [inline python/lsst/p_lsstSwig.i]: Why `%module`?  I don't think you want to do that --- it's declaring a new module, but this is supposed to be included by other modules.

Comment (timj) [inline python/lsst/p_lsstSwig.i]: You are right. I was probably a bit too trigger happy on copying the example code from http://swig.org/Doc3.0/Python.html#Python_2_unicode (albeit this code does actually solve the problem). I'll remove everything except for the `#define` and see what happens.

File: python/lsst/utils/multithreading/SharedData.py (hunk 2 above)
Comment (timj) [inline python/lsst/utils/multithreading/SharedData.py]: I think it's `not self._d` and I've made the change.

File: tests/raDecToStr.py (hunk 1 above)
Comment (timj) [inline tests/raDecToStr.py]: Message updated

File: python/lsst/p_lsstSwig.i (hunk 5 above)
Comment (timj) [inline python/lsst/p_lsstSwig.i]: I removed the module and I also added a test that fails without this unicode flag enabled.
